DigitalLearningSolutions.Data/Services/TutorialContentService.cs

```diff
@@ -39,62 +39,68 @@ int tutorialId
+                            AND Tutorials.ArchivedDate IS NULL
                             AND (
-                                 Tutorials.OrderByNumber < NextTutorials.OrderByNumber
-                                 OR Tutorials.TutorialID < NextTutorials.TutorialID
+                                 CustomisationTutorials.Status = 1
+                                 OR (CustomisationTutorials.DiagStatus = 1 AND Sections.DiagAssessPath IS NOT NULL AND Tutorials.DiagAssessOutOf > 0)
+                                 OR (Customisations.IsAssessed = 1 AND Sections.PLAssessPath IS NOT NULL)
                             )
-                            AND NextCustomisationTutorials.TutorialID = NextTutorials.TutorialID
+                  ),
+                  NextTutorial AS (
+                  SELECT TOP 1
+                         CurrentTutorial.TutorialID AS CurrentTutorialID,
+                         OtherTutorials.TutorialID AS NextTutorialID
+                    FROM Tutorials AS CurrentTutorial
+                         INNER JOIN OtherTutorials
+                         ON CurrentTutorial.SectionID = OtherTutorials.SectionID
 
-                         LEFT JOIN CustomisationTutorials AS NextSectionCustomisationTutorials
-                         ON NextSectionCustomisationTutorials.CustomisationID = @customisationId
-                            AND (
-                                 NextSectionCustomisationTutorials.Status = 1
-                                 OR NextSectionCustomisationTutorials.DiagStatus = 1
-                                 OR Customisations.IsAssessed = 1
-                            )
+                         INNER JOIN CustomisationTutorials AS OtherCustomisationTutorials
+                         ON OtherCustomisationTutorials.CustomisationID = @customisationId
+                            AND OtherCustomisationTutorials.Status = 1
 
-                         LEFT JOIN Tutorials AS NextSectionsTutorials
-                         ON NextSectionCustomisationTutorials.TutorialID = NextSectionsTutorials.TutorialID
-                            AND NextSectionsTutorials.ArchivedDate IS NULL
+                   WHERE CurrentTutorial.TutorialID = @tutorialId
+                         AND CurrentTutorial.OrderByNumber <= OtherTutorials.OrderByNumber
+                         AND (
+                              CurrentTutorial.OrderByNumber < OtherTutorials.OrderByNumber
+                              OR CurrentTutorial.TutorialID < OtherTutorials.TutorialID
+                         )
+                   ORDER BY OtherTutorials.OrderByNumber, OtherTutorials.TutorialID
+                  ),
+                  NextSection AS (
+                  SELECT TOP 1
+                         CurrentSection.SectionID AS CurrentSectionID,
+                         OtherTutorials.SectionID AS NextSectionID
+                    FROM Sections AS CurrentSection
+                         CROSS JOIN OtherTutorials

```

Comment:
Why is this join necessary? 🙂

Reply:
This picks up each of the other tutorials in the course (to then look at their section IDs). I've made this a bit shorter and tidier in 3f3f8da 🙂